py_pdf_parser/filtering.py

```diff
@@ -851,6 +852,71 @@ def remove_elements(self, *elements: "PDFElement") -> "ElementList":
             self.document, self.indexes - set(element._index for element in elements)
         )
 
+    def move_forwards_from(
+        self, element: "PDFElement", count: int = 1, capped: bool = False
+    ) -> "PDFElement":
+        """
+        Returns the element in the element list obtained by moving forwards from
+        `element` by `count`.
+
+        Args:
+            element (PDFElement): The element to start at.
+            count (int, optional): How many elements to move from `element`. The default
+                of 1 will move forwards by one element. Passing 0 will simply return the
+                element itself. You can also pass negative integers to move backwards.
+            capped (bool, optional): By default (False), if the count is high enough
+                that we try to move out of range of the list, an exception will be
+                raised. Passing `capped=True` will change this behaviour to instead
+                return the element at the start or end of the list.
+
+        Raises:
+            ElementOutOfRangeError: If the count large (or large-negative) enough that
+                we reach the end (or start) of the list. Only happens when capped=False.
+        """
+        indexes = sorted(self.indexes)
+        new_index = indexes.index(element._index) + count
+        if new_index < 0 or new_index >= len(indexes):
+            # Out of range. We could simply catch the index error for large new_index,
+            # but we have to handle the negative case like this anyway, so might as well
+            # do both cases while we're at it.
+            if capped:
+                new_index = max(min(new_index, len(indexes) - 1), 0)
+                element_index = indexes[new_index]
+                return self.document._element_list[element_index]
+            raise ElementOutOfRangeError(
+                f"Requested element is {'before' if capped < 0 else 'after'} the start "
```

Comment:
Shouldn't `new_index` be used here?
Also shouldn't it be `after the end of the ElementList`?

Reply:
@paulopaixaoamaral good spot... see latest commit